kansha/services/search/schema.py

```diff
@@ -167,3 +192,104 @@ def delta(cls, docid, **fields):
     def match(cls, v):
         '''Full text specific: match any of the fields'''
         return MATCHANYQuery(cls, v)
+
+
+class AltDocument(object):
+    '''Document Type for documents instanciated by a Schema object.'''
+
+    def __init__(self, schema, fields, docid, **field_values):
+        self._id = docid
+        self.fields = fields
+        self.schema = schema
+        for name, field in fields.iteritems():
+            setattr(self, name, field_values.get(name, field.default))
+
+    @property
+    def doc_type(self):
+        return self.schema
+
+
+class Schema(object):
+    '''Imperatively build a Document schema.
+
+    Same reserved keywords as the Declarative Schema.
+
+    Usage:
+
+        TestDocument = schema.Schema('MyDocument')
+        TestDocument.add_field('title', schema.TEXT(stored=True))
```

Comment:
"There should be one-- and preferably only one --obvious way to do it."

-> ``TestDocument.add_field(schema.TEXT('title', stored=True))``